src/lib/array.jl

```diff
@@ -160,15 +160,16 @@ end
 
 @adjoint getindex(i::Int, j::Int) = i[j], _ -> nothing
 
-function unzip(tuples)
-  map(1:length(first(tuples))) do i
-      map(tuple -> tuple[i], tuples)
-  end
+struct StaticGetter{i} end
+(::StaticGetter{i})(v) where {i} = v[i]
+@generated function _unzip(tuples, ::Val{N}) where {N}
+  Expr(:tuple, (:(map($(StaticGetter{i}()), tuples)) for i ∈ 1:N)...)
 end
```

Comment:
How about we get rid of the functor and just put the getindex into the source.
Does that end up with the same performance?

```suggestion
@generated function _unzip(tuples, ::Val{N}) where {N}
  args = ntuple(N) do ii
    :(map(x->x[$ii], tuples))
  end
  Expr(:tuple, args...)
end
```

Reply:
Did you try running that? You can't use closures inside a generated function (so can't do `x->x[$i]`). I did try some variations of this and always ended up making things worse. 

For example, the closest thing to your suggestion is:

```julia
@generated function _unzip(tuples, ::Val{N}) where {N}
  args = ntuple(N) do i
    :(map($(x -> x[i]), tuples))
  end
  Expr(:tuple, args...)
end

julia> @btime (x->last(map(+, fill(2x,100), fill(x,100))))'(0.1);
  31.180 μs (974 allocations: 36.31 KiB)
```

Reply:
> Did you try running that?

No, it was quite late for me by then.

> You can't use closures inside a generated function

I always forget this (even though it is a blocking issue for Arborist.jl).
I alway think it is that you can't call them in the body of the generated function, but can return them.